### DISCUSSION TITLE
Cargo linter custom target json?

### DISCUSSION THREAD

**Discussion by theoparis** [w0rp/ale#4065] [Q&A - Ask for help with problems]
I am building a kernel with rust and I need to specify a custom target json (`x86_64.json`) but I don't see a way to do this with the current cargo clippy linter. I am using neovim and I have no idea how I would add a custom linter myself in lua... I'd rather have the ability to change the cargo command through `vim.g`, but I only see options on configuring cargo features.

The only other option is to use VSCode 😕 

Do I need to make a github issue on adding custom targets instead?

If I don't specify a custom target it gives a ton of std errors. This also may be due to needing to use `cargo xclippy` instead of clippy, which automatically deals with non-standard library environments.

**Comment by hsanson** [comment]
I do not use rust tools so may not be of huge help here but can you post the whole ALEInfo output?

Not sure how to specify custom targets but maybe using g:ale_rust_cargo_clippy_options configuration can do? is in the help for ale-rust (e.g. :h ale-rust).
